packages/dds/map/src/localValues.ts

```diff
@@ -103,29 +103,34 @@ export class PlainLocalValue implements ILocalValue {
 export class LocalValueMaker {
 	/**
 	 * Create a new LocalValueMaker.
-	 * @param serializer - The serializer to serialize / parse handles.
 	 */
-	public constructor(private readonly serializer: IFluidSerializer) {}
+	public constructor() {}
 
 	/**
 	 * Create a new local value from an incoming serialized value.
 	 * @param serializable - The serializable value to make local
 	 */
-	// eslint-disable-next-line import/no-deprecated
-	public fromSerializable(serializable: ISerializableValue): ILocalValue {
+	public fromSerializable(
+		// eslint-disable-next-line import/no-deprecated
+		serializable: ISerializableValue,
+		serializer: IFluidSerializer,
+		bind: IFluidHandle,
+	): ILocalValue {
 		// Migrate from old shared value to handles
 		if (serializable.type === ValueType[ValueType.Shared]) {
 			serializable.type = ValueType[ValueType.Plain];
 			const handle: ISerializedHandle = {
 				type: "__fluid_handle__",
 				url: serializable.value as string,
 			};
-			serializable.value = handle;
+			// NOTE: here we require the use of `parseHandles` because the roundtrip
+			// through a string is necessary to resolve the absolute path of
+			// legacy handles (`ValueType.Shared`)
```

Comment:
I don't understand why the roundtrip through string is necessary. In this case, the incoming object `serializable` is a legacy format... as I think about it I don't understand how this can be. Or maybe worse - this may represent a gap in parsing handles in SharedObject. i.e. in your PR this is unreachable code (and this legacy handle wouldn't be parsed by SharedOject).

Aside from this case, my assumption is all the objects in this package have real handles. Because we are now parsing incoming handles in SharedObject.  BuT this is apparently a case where the generic `parseHandle` logic isn't sufficient.

So we may need to inject this backcompat code into the SharedObject thing.  i.e. DDSes get to provide a callback in case there's custom handle conversion logic required.

Reply:
Ok Connor set me straight offline:

> this legacy handle wouldn't be parsed by SharedOject

This is true, and hence we will still get to this codepath. So no, "this is unreachable code" is not correct.

And he explained how in this legacy case the handle path may be relative, and depends on some patch-up code in serializer's `decode` logic to switch to absolute path.